fe/fe-core/src/main/java/com/starrocks/connector/ConnectorFactory.java

```diff
@@ -0,0 +1,18 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector;
+
+public interface ConnectorFactory {
+    /**
+     * create a connector instance
+     * @param context - encapsulate all information needed to create a connector
+     * @return a connector instance
+     */
+    Connector createConnector(Connector.Context context);
```

Comment:
I think it could exist catalogName in connector

Reply:
can we add it later when we really need it?

Reply:
mou man tai, I will update this part in follow days.

Reply:
but the `Context` is really necessary?  only configs in the context.

Reply:
I think when we support spi, there are more parameter we need to pass to connector factory

Reply:
Properties is enough. xxxConnector will initialize themself field by parsing the properties. that's all. 
The syntax for creating catalog is only catalog name and properties.

Reply:
The main problem is that I don't think there need to be too many additional filed in `catalog`.

Reply:
if so, why do we need a catalog